rule_manager.py

```diff
@@ -0,0 +1,135 @@
+from typing import Any, Dict, Optional
+
+from pypanther import LogType, Rule, get_panther_rules, get_rules, register
+from pypanther.wrap import exclude, include
+
+
+class RuleManager:
+    def __init__(self):
+        self.rules: Dict[str, Rule] = {}
+        self.panther_rules: Dict[str, Rule] = {}
+        self.custom_rules: Dict[str, Rule] = {}
+
+    def load_managed_rules(self, **kwargs):
+        """Load Panther-managed rules."""
+        panther_rules_list = get_panther_rules(**kwargs)
+        self.panther_rules = {rule.id: rule for rule in panther_rules_list}
+        self.rules.update(self.panther_rules)
```

Comment:
Interestingly this means that when you:
- Call `load_managed_rules` the first time, the `panther_rules` and `rules` dictionaries are the same.
- If you later call `load_managed_rules` with input that returns fewer rules, then the `panther_rules` dictionary will include only those rules but the `rules` dictionary will include those and any old ones that were there from the first call

I understand the intent is to not have `load_managed_rules` and `load_custom_rules` step on each other's toes but I think there needs something slightly more complicated here.